app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-finder-tracker.js

```diff
@@ -46,76 +46,9 @@
-          wasFilterRemoved = !eventTarget.checked
-          break
-        }
-        case 'radio': {
-          const radioId = eventTarget.id
-
-          // The "value" we need for a radio is the label text that the user sees beside the checkbox.
-          elementValue = document.querySelector("label[for='" + radioId + "']").textContent
-          defaultValue = eventTarget.closest('[data-ga4-section]').querySelector('input[type=radio]:first-of-type')
-
-          if (eventTarget.id === defaultValue.id) {
-            // Radio elements being reverted to their first option (i.e. their default value) count as a "removed filter".
-            wasFilterRemoved = true
-          }
-          break
-        }
-        case 'select':
-          // The value of a <select> is the value attribute of the selected <option>, which is a hyphenated key. We need to grab the human readable label instead for tracking.
-          elementValue = eventTarget.querySelector("option[value='" + eventTarget.value + "']").textContent
-          defaultValue = eventTarget.querySelector('option:first-of-type').textContent
-
-          if (elementValue === defaultValue) {
-            // <select> elements being reverted to their first option (i.e. their default value) count as a "removed filter". (This will be used on the filter <select>s but not the sort by <select>, as you can't "remove" the sort by filter.)
-            wasFilterRemoved = true
-          }
-          break
-
-        case 'text':
-          elementValue = eventTarget.value
-          if (elementValue === '') {
-            // If our custom date filters are reset, they become an empty text box, so we count this as a "removed filter". This boolean won't be used for the keyword search box, as deleting the keyword isn't considered removing a filter.
-            wasFilterRemoved = true
-          }
-          break
-
-        case 'date': {
-          // The GOV.UK Design System date input consists of three grouped but separate fields (day,
-          // month, year). We want to fire a single event when all three fields are filled in to
-          // avoid firing excessive events.
-          const inputs = [...eventTarget.closest('.govuk-date-input').querySelectorAll('input')]
-          const allInputsSet = inputs.every(input => input.value)
-          const noInputsSet = inputs.every(input => !input.value)
-
-          if (allInputsSet) {
-            elementValue = inputs.map(input => input.value).join('/')
-          } else if (noInputsSet) {
-            wasFilterRemoved = true
-          } else {
-            // Do not track partially filled in fields
-            return null
-          }
-          break
-        }
-      }
-
-      return { elementValue: elementValue, wasFilterRemoved: wasFilterRemoved }
+      return supportedElements[elementType] ? supportedElements[elementType](event.target, event) : { elementValue: '', wasFilterRemoved: false }
```

Comment:
Does `event` need to be passed here? Is this just for future elements to have access to it if they need it?

Reply:
Yeah, this is for when an `getElementType` function needs to access some information in the event. If you refer to `creates the correct GA4 object for an additionally specified element` in the spec, you can see an example of this.